jupyter-extension/jupyterlab_pachyderm/handlers.py

```diff
@@ -240,15 +267,47 @@ async def get(self, path):
         self._finish_model(model, location=False)
 
 
+# TODO: see about writing to/from config file
 class ConfigHandler(BaseHandler):
+    def config_response(self) -> bytes:
+        if not self.client:
+            return json.dumps({"cluster_status": "INVALID"})
+
+        try:
+            self.client.auth.who_am_i()
+            cluster_status = "AUTH_ENABLED"
+        except grpc.RpcError as err:
+            if err.code() == grpc.StatusCode.UNAUTHENTICATED:
+                cluster_status = "AUTH_ENABLED"
+            elif (
+                err.code() == grpc.StatusCode.UNIMPLEMENTED
+                and "the auth service is not activated" in err.details()
+            ):
+                cluster_status = "AUTH_DISABLED"
+            else:
+                cluster_status = "INVALID"
+        except ConnectionError:
+            cluster_status = "INVALID"
+
+        return json.dumps(
+            {"cluster_status": cluster_status, "pachd_address": self.client.address}
+        )
+
     @tornado.web.authenticated
     async def put(self):
         try:
             body = self.get_json_body()
-            response = await self.mount_client.config(body)
+            address: str = body["pachd_address"]
+
+            if address.removeprefix("grpc://") != self.client.address:
```

Comment:
2 notes:
1) There are a few more URL schemes that pachd addresses can use. We can copy the logic (here)[https://github.com/pachyderm/pachyderm/blob/master/src/internal/grpcutil/addr.go#L46] to account for them.
2) If users attempt to connect to a secure endpoint (https or grpcs), they'll need to provide a certificate. The user inputs the cert in the advanced settings on the cluster connect page in the extension. The certificate is used to create an encrypted channel between the client and server. The certificate needs to be passed to the Python client; you can see (here)[https://github.com/pachyderm/pachyderm/blob/master/src/server/pfs/fuse/server.go#L1070] how we pass the certs to the Go client that the mount server uses.

Reply:
For (2), do you have a vision for how the user would supply these certs? I wasn't aware we were currently able to do this with the FUSE version

Reply:
On the config page in the extension where you enter your cluster address, there's an advanced settings button that allows you to enter the cert. We then pass it along with the connect request to the server.

Reply:
thanks, i missed the certs option. i've added it in now.

Reply:
Bump point 1 above

Reply:
sorry, somehow I missed this comment. what is written here should be equivalent to what we are doing in the mount-server. `removeprefix` leaves the string intact if the prefix is not found. `grpc` addresses have the `grpc` removed in the `address` property of the client, but for others the protocol is contained within the address. those protocols should still be handled correctly by the client.